plot-lib/plot/private/utils-and-no-gui.rkt

```diff
@@ -21,9 +21,12 @@
  Legend-Anchor)
 
 (require "common/plotmetrics.rkt")
+(require/typed (submod "common/plotmetrics.rkt" untyped)
+  [#:opaque plot-metrics<%> interface?])
```

Comment:
Hi @samth , could you provide some advice on how to export an interface from Typed Racket?   `plot-metrics<%>` is an interface, and I would like to be exported and used from plain Racket when the user requires the `plot` or `plot/no-gui` package.

The code above exports `plot-metrics<%>` but it is a type, and trying to do a `(is-a? o plot-metrics<%>)` fails with "type-check: type name used out of context".   For an example of this failure, see "plot-test/plot/tests/PRs/90.rkt".

Thanks,
Alex.

Reply:
This was the first thing I tried, and didn't work.  On further investigation, it seems that the problem is that the line `(provide (all-from-out "private/utils-and-no-gui.rkt"))`  in "utils.rkt" does not actually export the `plot-metrics<%>` interface.  However, adding an explicit provide there fixed the problem.

There are some other problems with the way that interface was defined, I think I will fix them as a separate pull request, to keep this one as a "documentation only" PR.